samples/apps/copilot-chat-app/webapi/Skills/ChatSkill.cs

```diff
@@ -154,12 +178,48 @@ await foreach (var memory in results)
 
         // Update the token limit.
         memoryText = $"Past memories (format: [memory type] <label>: <details>):\n{memoryText.Trim()}";
-        tokenLimit -= Utils.TokenCount(memoryText);
+        tokenLimit -= Utilities.TokenCount(memoryText);
         context.Variables.Set("tokenLimit", tokenLimit.ToString(new NumberFormatInfo()));
 
         return memoryText;
     }
 
+    /// <summary>
+    /// Extract relevant additional knowledge using a Semantic Kernel planner.
+    /// </summary>
+    [SKFunction("Acquire external information")]
+    [SKFunctionName("AcquireExternalInformation")]
+    [SKFunctionContextParameter(Name = "userIntent", Description = "The intent of the user.")]
+    [SKFunctionContextParameter(Name = "tokenLimit", Description = "Maximum number of tokens")]
+    public async Task<string> AcquireExternalInformationAsync(SKContext context)
+    {
+        if (!this._plannerOptions.Enabled)
+        {
+            return string.Empty;
+        }
+
+        // Create a plan and run it.
+        SKContext planContext = await (await (await this._plannerFactoryAsync(this._kernel))
+            .CreatePlanAsync(context["userIntent"]))
+            .InvokeAsync(context);
```

Comment:
What is `context.Input` ? Are you sure you always want that as the input to the plan execution?